### ISSUE TITLE
Add option to find unique elements in nested sequences when calling `Dataset.unique`

### ISSUE THREAD

**Issue by mariosasko** [huggingface/datasets#3299]
It would be nice to have an option to flatten nested sequences to find unique elements stored in them when calling `Dataset.unique`. Currently, `Dataset.unique` only supports finding unique sequences and not unique elements in that situation.

**Comment by sayakpaul** [comment]
Hi @mariosasko!

Has this been patched into any of the releases?

**Comment by dcruiz01** [comment]
@mariosasko did this ever get implemented? Willing to help if you are still up for it.

**Comment by mariosasko** [comment]
@dcruiz01 No, but here is an example of how to do this with the existing API:


```python
ds = Dataset.from_dict({"tokens": [["a", "b"], ["c", "a"], ["c", "e"]]})

def flatten_tokens(pa_table):
    return pa.table([pc.list_flatten(pa_table["tokens"])], ["flat_tokens"])

ds = ds.with_format("arrow")
ds = ds.map(flatten_tokens, batched=True)
ds = ds.with_format(None)

unique_tokens = ds.unique("flat_tokens")
```

When I think about it, `.unique` on `Sequence(Value(...))` should return unique sequences/arrays, not unique elements of these sequences...
